colour/recovery/meng2015.py

```diff
@@ -177,12 +176,12 @@ def XYZ_to_sd_Meng2015(
 
     sd = sd_ones(cmfs.shape)
 
-    def objective_function(a: ArrayLike) -> NDArrayFloat:
+    def objective_function(a: NDArrayFloat) -> DTypeFloat:
         """Define the objective function."""
 
-        return cast(NDArrayFloat, np.sum(np.diff(a) ** 2))
+        return np.sum(np.square(np.diff(a)))
```

Comment:
In `numpy==2.2.0` there was a bug that caused `np.diff(a) ** 2` to be inferred as `np.signedinteger`. This has been resolved in `2.2.4`, but I figured that `square` would be a more stable approach.